packages/node_modules/@webex/plugin-meetings/src/stats/metrics.js

```diff
@@ -0,0 +1,164 @@
+    super({}, options);
+    /**
+     * @instance
+     * @type {Object}
+     * @readonly
+     * @private
+     * @memberof MediaMetrics
+     */
+    this.attrs = attrs;
+    /**
+     * @instance
+     * @type {Object}
+     * @readonly
+     * @private
+     * @memberof MediaMetrics
+     */
+    this.options = options;
+    /**
+     * @instance
+     * @type {MeetingStats}
+     * @readonly
+     * @private
+     * @memberof MediaMetrics
+     */
+    this.stats = null;
+  }
+
+  /**
+   * @param {MeetingStats} stats
+   * @returns {undefined}
+   * @memberof MediaMetrics
+   */
+  setStats(stats) {
+    this.stats = stats;
+  }
+
+  /**
+   * checks to see if we can send data based on the config interval
+   * emits an event if so
+   * @param {String} id
+   * @param {MeetingStats} sendRecv
+   * @returns {undefined}
+   * @private
+   * @memberof MediaMetrics
+   */
+  checkData(id, sendRecv) {
+    const metrics = sendRecv.getMetrics();
+
+    if (metrics.get().length >= this.config.stats.mediaMetricsInterval) {
+      // we could technically parse this data down with the analyzer, but don't want to throw data away
+      const interval = metrics.getSlice(this.config.stats.mediaMetricsInterval);
+
+      Trigger.trigger(
+        this,
+        {
+          file: 'stats/metrics',
+          function: 'senderData'
+        },
+        EVENT_TRIGGERS.METRICS_MEDIA_PAYLOAD,
+        {
+          id,
+          interval
+        }
+      );
+      metrics.clear();
+    }
+  }
+
+  /**
+   * sets up all the default senders and receivers getStats to collect data
+   * @returns {undefined}
+   * @public
+   * @memberof MediaMetrics
+   */
+  initialize() {
+    return {
+      useConfig: true,
+      senders: [
+        {
+          id: STATS.audioSenderId,
+          correlate: STATS.audioCorrelate,
+          metrics: true,
+          onData: () => {
+            this.checkData(STATS.audioSenderId, this.stats.getSender(STATS.audioSenderId));
+          }
+        },
+        {
+          id: STATS.videoSenderId,
+          correlate: STATS.videoCorrelate,
+          metrics: true,
+          onData: () => {
+            this.checkData(STATS.videoSenderId, this.stats.getSender(STATS.videoSenderId));
+          }
+        },
+        {
+          id: STATS.shareSenderId,
+          correlate: STATS.shareCorrelate,
+          metrics: true,
+          onData: () => {
+            this.checkData(STATS.shareSenderId, this.stats.getSender(STATS.shareSenderId));
+          }
+        }
+      ],
+      receivers: [
+        {
+          id: STATS.audioReceiverId,
+          correlate: STATS.audioCorrelate,
+          metrics: true,
+          onData: () => {
+            this.checkData(STATS.audioReceiverId, this.stats.getReceiver(STATS.audioReceiverId));
+          }
+        },
+        {
+          id: STATS.videoReceiverId,
+          correlate: STATS.videoCorrelate,
+          metrics: true,
+          onData: () => {
```

Comment:
I'm guessing we don't control the interval for getStats() so we have to do the check on every event message?

Reply:
well, they are separate between getStats and the metrics needed for clients, and can be configured. getStats can be done entirely independently, so short answer yes.